### DISCUSSION TITLE
使用官方的lua，导致[:00000000] A message from [ :01000002 ] to [ :01000002 ] maybe in an endless loop (version = 1)

### DISCUSSION THREAD

**Discussion by liyqxtu** [cloudwu/skynet#1802] [General]
个人建议，修改官方lua的源码，也还是应该兼容官方lua，毕竟官方lua才是大多数人的选择
[:00000000] A message from [ :01000002 ] to [ :01000002 ] maybe in an endless loop (version = 1)
[:00000000] A message from [ :01000002 ] to [ :01000002 ] maybe in an endless loop (version = 1)
[:00000000] A message from [ :01000002 ] to [ :01000002 ] maybe in an endless loop (version = 1)
[:00000000] A message from [ :01000002 ] to [ :01000002 ] maybe in an endless loop (version = 1)

**Comment by cloudwu** [comment]
光看 log 无法确定原因，请先定位原因。

用官方 lua 启动 example 并无问题，skynet 也并未修改 lua 的任何特性。

**Comment by liyqxtu** [comment]
请教一下大佬
1、调试的时候，是不是应该把这个endless loop关掉呀？有没有什么开关之类的东西呢？
2、自己电脑上设置的lua_path，在运行skynet后，就不起作用了，是在哪里被覆盖掉了么？
3、具体问题如下 [:01000005] init service failed: ./service/cslave.lua:237: Can't connect to master
![1](https://github.com/cloudwu/skynet/assets/4734699/78bcc289-57ca-4bc1-9bf5-bae84afbf934)

**Comment by liyqxtu** [comment]
个人愚见，从功能上讲，skynet_error 这个函数是在记录日志，而不是记录错误，是不是改成 skynet_log 可能更合理一点呢？

**Comment by cloudwu** [comment]
你可以自定义 log 服务，过滤你想过滤的信息。

**Comment by terry8210** [comment]
signal address sig 向服务发送一个信号，sig 默认为 0 。当一个服务陷入死循环时，默认信号会打断正在执行的 lua 字节码，并抛出 error 显示调用栈。这是针对 endless loop 的 log 的有效调试方法。注：这里的信号并非系统信号。
